### PR TITLE
Switch to a tsan-instrumented libc++ for tsan tests

### PR DESCRIPTION
**Pull request by dmitri-d**
This fixes https://github.com/envoyproxy/envoy/issues/9784 and re-enables vhds_integration_test

Signed-off-by: Dmitri Dolguikh <ddolguik@redhat.com>

Risk Level: Low, but will most likely increase memory usage

### CODE REVIEW

**Comment by dmitri-d** [comment]
ping @lizan, @alyssawilk

File: test/integration/BUILD [hunk 1]
```diff
@@ -220,7 +220,6 @@ envoy_cc_test(
     tags = [
         "fails_on_windows",
         # https://github.com/envoyproxy/envoy/issues/9784
```
Comment (lizan) [inline test/integration/BUILD:222]: delete this line too

**Comment by dmitri-d** [comment]
* responded to feedback

**Comment by dmitri-d** [comment]
@lizan: just realised I haven't switched ci to use instrumented libc++, updated.

**Comment by dmitri-d** [comment]
I think I've seen the failure in the xds_integration_test before, but it looks to me that there might be a race there -- we turn off stream synchronization (here: https://github.com/envoyproxy/envoy/blob/master/source/common/filesystem/posix/filesystem_impl.cc#L99).

**Comment by lizan** [comment]
Hmm, I don't think we should call that there. It has nothing to do with the fstream actually. Can you just delete that line?

**Comment by dmitri-d** [comment]
* iostreams stay synchronised now in InstanceImplPosix::fileReadToEnd

**Comment by dmitri-d** [comment]
* merged in latest changes: format check failed on files untouched by this PR

**Comment by lizan** [comment]
master is failing, don't retry

**Pull request merged**